src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2676,7 +2676,7 @@ private BoundStatement BindReturn(ReturnStatementSyntax syntax, DiagnosticBag di
                 }
             }
 
-            return new BoundReturnStatement(syntax, refKind, arg);
+            return new BoundReturnStatement(syntax, sigRefKind, arg);
```

Comment:
💭 This seems to be a bug. The bound return statement for an explicit return statement was not behaving the same way as the bound return statement for an expression-bodied member. This change updates the explicit return statement behavior to use the ref kind of the method signature, which matches expression-bodied members.